src/core/thread/mesh_forwarder.cpp

```diff
@@ -806,12 +806,15 @@ Neighbor *MeshForwarder::UpdateNeighborOnSentFrame(Mac::TxFrame &aFrame, otError
     else if (aError == OT_ERROR_NO_ACK)
     {
         neighbor->IncrementLinkFailures();
+
+#if OPENTHREAD_FTD
```

Comment:
Let me double-check if we do not need this on MTD?
For example I think a SED child not getting acks from its parent would go through this code path... and then would remove its parent (which would cause it to detach)?

Reply:
Yes, you are correct. I'll change it to be based on role.